src/applications/check-in/components/pages/Appointments/index.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import { useFormRouting } from '../../../hooks/useFormRouting';
+import { useGetCheckInData } from '../../../hooks/useGetCheckInData';
+import Wrapper from '../../layout/Wrapper';
+import { APP_NAMES } from '../../../utils/appConstants';
+import { makeSelectApp } from '../../../selectors';
+import { useUpdateError } from '../../../hooks/useUpdateError';
+
+const AppointmentsPage = props => {
+  const { router } = props;
+  const selectApp = useMemo(makeSelectApp, []);
+  const { app } = useSelector(selectApp);
+  const { t } = useTranslation();
+  const [isLoading, setIsLoading] = useState(true);
+  const { goToNextPage } = useFormRouting(router);
+  const { updateError } = useUpdateError();
+
+  const {
+    isComplete,
+    refreshCheckInData,
+    checkInDataError,
+  } = useGetCheckInData({
+    refreshNeeded: true,
+    router,
+    isPreCheckIn: app === APP_NAMES.PRE_CHECK_IN,
+  });
+
+  const handleClick = () => {
+    goToNextPage();
+  };
+
+  useEffect(
+    () => {
+      setIsLoading(!isComplete);
+      if (!isComplete) {
+        refreshCheckInData();
+      }
+    },
+    [isComplete, refreshCheckInData],
+  );
+
+  // TODO: Remove this and the checkInDataError property from the useGetCheckInData hook
+  // We could call updateError in the hook
+  useEffect(
+    () => {
+      if (checkInDataError) {
+        updateError('error-fromlocation-precheckin-introduction');
+      }
+    },
+    [checkInDataError, updateError],
+  );
+
+  if (isLoading) {
+    window.scrollTo(0, 0);
+    return (
+      <div>
+        <va-loading-indicator
+          data-testid="loading-indicator"
+          message={t('loading-upcoming-appointments')}
```

Comment:
we should double check with UX on any new text content. I would default to using an existing loading message. 
```  "loading-your-appointment-details": "Loading your appointment details",
  "loading-your-appointments-for-today": "Loading your appointments for today",
  "loading-your-check-in-experience": "Loading your check in experience",```

Reply:
I used 'loading-your-appointment-details'